scala-lang/src/test/scala/com/baeldung/scala/callbynameandvalue/CallByNameCallByValueTest.scala

```diff
@@ -0,0 +1,35 @@
+package com.baeldung.scala.callbynameandvalue
+
+import com.baeldung.scala.callbynameandvalue.CallByNameCallByValue._
+import org.scalatest.{FlatSpec, Matchers}
+
+/**
+ * @author vid2010
+ */
+class CallByNameCallByValueTest extends FlatSpec with Matchers {
+
+  "A call by-value" should "have evaluated before passing an argument" in {
+    val currentTime = System.currentTimeMillis()
+    getTimeByVal(currentTime) should be(currentTime)
+  }
+
+  "A call by-name" should "have evaluated when an argument used inside the function's body" in {
+    getTimeByName(System.currentTimeMillis()) should not be (System.currentTimeMillis())
```

Comment:
Let's save the current time as a variable here too, for the same reason as above.

Reply:
Hi @glmartin,
If I declare the current time as a variable here the test case will fail because for by-name parameter the value of System.currentTimeMillis() will evaluate before passing.

Reply:
I see. Thanks.